vllm/engine/llm_engine.py

```diff
@@ -754,9 +757,17 @@ def has_unfinished_requests_for_virtual_engine(
     def _process_sequence_group_outputs(
         self,
         seq_group: SequenceGroup,
-        outputs: List[EmbeddingSequenceGroupOutput],
+        outputs: Union[List[EmbeddingSequenceGroupOutput],
+                       List[SimpleSequenceGroupOutput]],
+        model_mode: ModelMode,
     ) -> None:
-        seq_group.embeddings = outputs[0].embeddings
+        if model_mode == ModelMode.EMBEDDING:
+            seq_group.embeddings = outputs[0].embeddings
+        elif model_mode == ModelMode.SIMPLE:
+            seq_group.result = outputs[0].result
+        else:
+            raise ValueError(
+                "Only support ModelMode.EMBEDDING Or ModelMode.SIMPLE")
```

Comment:
I wonder whether we could also factor out this logic into the individual `ModelMode`s....